react/src/users/UsersContainer.js

```diff
@@ -1,22 +1,62 @@
 import { withRouter } from 'react-router-dom';
 import axios from 'axios';
-import CircularProgress from '@mui/material/CircularProgress';
+import { CircularProgress, Alert } from '@mui/material';
 import React, { useState, useEffect } from 'react';
 import ROUTES_USERS from './routes';
 import UsersPage from './components/UsersPage';
 
-const UsersContainer = () => {
+const UsersContainer = (props) => {
+  const { currentUser } = props;
   const [loading, setLoading] = useState(false);
   const [users, setUsers] = useState([]);
-  const [userUpdates, setUserUpdates] = useState([]);
-
+  const [newUser, setNewUser] = useState('');
+  const [userUpdates, setUserUpdates] = useState({});
+  const [permissionMessage, setPermissionMessage] = useState('');
+  const [permissionMessageSeverity, setPermissionMessageSeverity] = useState('');
   const refreshDetails = () => {
     setLoading(true);
     axios.get(ROUTES_USERS.LIST).then((listResponse) => {
       setUsers(listResponse.data);
     });
   };
 
+  const handleAddNewUser = () => {
+    axios.post(ROUTES_USERS.CREATE, { idir: newUser }).then((response) => {
+      if (response.status === 200) {
+        refreshDetails();
```

Comment:
Instead of calling `refreshDetails()`, perhaps the response can include the newly created user and their permissions, and we can update the `users` state using Immer (https://immerjs.github.io/immer/)?